x/confidentialtransfers/types/transfer.go

```diff
@@ -205,6 +215,113 @@ func NewTransfer(
 	}, nil
 }
 
+// Decrypts the transfer transaction and returns the decrypted data.
+// The aesKey and decryptAvailableBalance field are only used if address is the sender address of the transaction.
+func (r *Transfer) Decrypt(decryptor *elgamal.TwistedElGamal, privKey ecdsa.PrivateKey, decryptAvailableBalance bool, address string) (*TransferDecrypted, error) {
```

Comment:
I think generalization in a form of `Decryptable` interface, leads to more complexity in current implementation and/or degraded implementation (like skipped address params in other implementation).
We may need type (sender/receiver/auditor) based implementations of `Decryptable` and a separate factory function.

Reply:
Thanks for the feedback - since we figured out how to parse the message type properly, I removed the Decryptable interface and just have it call Decrypt for each method.

Also split the Decrypt logic up in the Transfer class to shorten the function.